ARCHITECTURE.md

```diff
@@ -0,0 +1,207 @@
+# Architectural Tenets
+
+<p style="font-size: 120%;">
+  <b>TL;DR:</b> work done at code.org should live in the code-dot-org monorepo, be deployed in a container as part of one of our existing monoliths, be integration tested by our central CI system prior to PR merge, use REST for client/server communication, default to "The Rails Way", prefer 3rd party managed services that implement Open Source Protocols/APIS, and be implemented using Ruby/Rails/JS/TS/React/MySQL.
+</p>
+
+1. If somebody proposes a design counter to these tenets, please direct them to [this link](ARCHITECTURE.md#architectural-tenets) to this section of this file. Better yet: link them straight to the relevant tenet(s) below.
+2. If these tenets become outdated, they should either be amended publicly or deleted.
+3. If this file is still in the repo, its rules are still in effect.
+
+The goal of the tenets is not to maintain an exhaustive or  current list of technical requirements, but to highlight a small set of **core architectural choices we rarely break**  and rarely change. These architectural tenets are intentionally limited: they were mostly true in 2014, are mostly true today, and might even end up mostly true in 2034. Think “constitution” not “laws”: its important they are amended to fit the times, but only rarely. Less is more.
+
+## The tenets represent tradeoffs
+
+The tenets may be different than you are familiar with in commercial orgs where unbounded staff growth can often be assumed. They work together so our permanently small team can make technical progress both today AND in the long-term. They balance inherent tradeoffs between “personal dev velocity today” and “collective dev velocity years from now”.
+
+Examples:
+
+| Collective Benefit Tomorrow | Trade Off Today |
+| :---- | :---- |
+| **“If my PR passes  CI it won’t break code.org”** | “CI is slow” |
+| **“I  can implement features/refactors that require changes across the codebase: even in areas I didn’t know about before (even if nobody knows about them anymore)”** | “I have to work inside the monorepo even if it’d be so much faster to deliver this quarter’s feature goals by spinning out my own repo without the overhead” |
+| **“After mastering a few key technologies I can work in almost any part of the codebase”** | “Even if hotnewframework is the very best option in 2120, it might be best if I use existingokframework from 2115 because we already use it widely and it gets the job done just fine.” |
+
+Historical violations of almost all tenets exist, and are de facto grandfathered in. However, these should not be interpreted as an invitation to follow suit. A strong majority of historical code already adheres to all the tenets, which is part of why they were picked in the first place.
```

Comment:
```suggestion
Historical violations of almost all tenets exist, and are not a priority to fix. However, these should not be interpreted as an invitation to follow suit. A strong majority of historical code already adheres to all the tenets, which is part of why they were picked in the first place.
```

rewriting to avoid use of a loaded term